arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/ParTupledN.kt

```diff
@@ -272,3 +277,179 @@ suspend fun <A, B, C, D> parMapN(
+ *
+ * **WARNING** this function forks [fa], [fb], [fc], [fd], [fe], [ff], [fg] but if it runs in parallel depends
+ * on the capabilities of the provided [CoroutineContext].
+ * We ensure they start in sequence so it's guaranteed to finish on a single threaded context.
+ *
+ * @see parMapN for a function that ensures it runs in parallel on the [ComputationPool].
+ */
+suspend fun <A, B, C, D, E, F, G, H> parMapN(
+  ctx: CoroutineContext,
+  fa: suspend () -> A,
+  fb: suspend () -> B,
+  fc: suspend () -> C,
+  fd: suspend () -> D,
+  fe: suspend () -> E,
+  ff: suspend () -> F,
+  fg: suspend () -> G,
+  f: (A, B, C, D, E, F, G) -> H
+): H =
+  parMapN(
+    ctx,
+    suspend { parMapN(ctx, fa, fb, fc, fd, fe, ff, ::Tuple6) },
+    fg,
+  ) { abcdef, g ->
+    val (a, b, c, d, e, _f) = abcdef
+    f(a, b, c , d, e, _f, g)
+  }
+
+/**
+ * Parallel maps [fa], [fb], [fc], [fd], [fe], [ff], [fg], [fh] on the provided [CoroutineContext].
+ * Cancelling this operation cancels both tasks running in parallel.
+ *
+ * **WARNING** this function forks [fa], [fb], [fc], [fd], [fe], [ff], [fg], [fh] but if it runs in parallel depends
+ * on the capabilities of the provided [CoroutineContext].
+ * We ensure they start in sequence so it's guaranteed to finish on a single threaded context.
+ *
+ * @see parMapN for a function that ensures it runs in parallel on the [ComputationPool].
+ */
+suspend fun <A, B, C, D, E, F, G, H, I> parMapN(
+  ctx: CoroutineContext,
+  fa: suspend () -> A,
+  fb: suspend () -> B,
+  fc: suspend () -> C,
+  fd: suspend () -> D,
+  fe: suspend () -> E,
+  ff: suspend () -> F,
+  fg: suspend () -> G,
+  fh: suspend () -> H,
+  f: (A, B, C, D, E, F, G, H) -> I
+): I =
+  parMapN(
+    ctx,
+    suspend { parMapN(ctx, fa, fb, fc, fd, fe, ff, fg, ::Tuple7) },
+    fh
+  ) { abcdefg, h ->
+    val (a, b, c, d, e, _f, g) = abcdefg
+    f(a, b, c , d, e, _f, g, h)
+  }
+/**
+ * Parallel maps [fa], [fb], [fc], [fd], [fe], [ff], [fg], [fh], [fi] on the provided [CoroutineContext].
+ * Cancelling this operation cancels both tasks running in parallel.
+ *
+ * **WARNING** this function forks [fa], [fb], [fc], [fd], [fe], [ff], [fg], [fh], [fi] but if it runs in parallel depends
+ * on the capabilities of the provided [CoroutineContext].
+ * We ensure they start in sequence so it's guaranteed to finish on a single threaded context.
+ *
+ * @see parMapN for a function that ensures it runs in parallel on the [ComputationPool].
+ */
+suspend fun <A, B, C, D, E, F, G, H, I, J> parMapN(
+  ctx: CoroutineContext,
+  fa: suspend () -> A,
+  fb: suspend () -> B,
+  fc: suspend () -> C,
+  fd: suspend () -> D,
+  fe: suspend () -> E,
+  ff: suspend () -> F,
+  fg: suspend () -> G,
+  fh: suspend () -> H,
+  fi: suspend () -> I,
+  f: (A, B, C, D, E, F, G, H, I) -> J
+): J =
+  parMapN(
+    ctx,
+    suspend { parMapN(ctx, fa, fb, fc, fd, fe, ff, fg, fh) { a, b, c, d, e, _f, g, h -> Tuple8(a, b, c, d, e, _f, g, h) } },
```

Comment:
I'm not sure if I'm just having trouble with my IDE or what but it doesn't appear to work with just `::Tuple8` here